facia-tool/public/js/models/collections/collection.js

```diff
@@ -319,6 +319,25 @@ define([
         return this.front.mode() !== 'treats' && this.history().length;
     };
 
+    Collection.prototype.replaceArticle = function(articleId) {
+        var self = this;
+        var collectionList = this.front.getCollectionList(this.raw);
+
+        _.each(collectionList, function(item, index) {
```

Comment:
We can avoid looping on all articles using `_.find()` instead of `_.each` to get `item`.